docs/03-tutorials/00-api-exposure/apix-04-expose-and-secure-workload-jwt.md

```diff
@@ -0,0 +1,150 @@
+---
+title: Expose and secure a workload with JWT
+---
+
+This tutorial shows how to expose and secure services or Functions using API Gateway Controller. The Controller reacts to an instance of the API Rule custom resource (CR) and creates an Istio Virtual Service and [Oathkeeper Access Rules](https://www.ory.sh/docs/oathkeeper/api-access-rules) according to the details specified in the CR. To interact with the secured workloads, the tutorial uses a JWT token.
+
+You can use it as a follow-up to the [Use a custom domain to expose a workload](./apix-01-own-domain.md) tutorial.
+
+## Prerequisites
+
+This tutorial is based on a sample HttpBin service deployment and a sample Function. To deploy or create them, follow the [Create a workload](./apix-02-create-workload.md) tutorial.
+
+## Get a JWT access token
+
+1. In your OpenID Connect-compliant (OIDC-compliant) identity provider, create an application to get your client credentials such as Client ID and Client Secret. Export your client credentials as environment variables. Run:
+
+   ```bash
+   export CLIENT_ID={YOUR_CLIENT_ID}
+   export CLIENT_SECRET={YOUR_CLIENT_SECRET}
+   ```
+
+2. Encode your client credentials and export them as an environment variable:
+
+   ```bash
+   export ENCODED_CREDENTIALS=$(echo -n "$CLIENT_ID:$CLIENT_SECRET" | base64)
+   ```
+
+3. In your browser, go to `https://YOUR_OIDC_COMPLIANT_IDENTITY_PROVIDER_INSTANCE/.well-known/openid-configuration`, save values of the **token_endpoint** and **jwks_uri** parameters, and export them as environment variables:
+
+   ```bash
+   export TOKEN_ENDPOINT={YOUR_TOKEN_ENDPOINT}
+   export JWKS_URI={YOUR_JWKS_URI}
+   ```
+
+4. Get the JWT access token:
+
+   ```bash
+   curl -X POST "$TOKEN_ENDPOINT" -d "grant_type=client_credentials" -d "client_id=$CLIENT_ID" -H "Content-Type: application/x-www-form-urlencoded" -H "Authorization: Basic $ENCODED_CREDENTIALS"
```

Comment:
Actually this curl can vary because the request has to match to what provider expects. In my case when I used [auth0](https://auth0.com) I had to put `audience` parameter in the request because my configuration had no default for this property. How should we tackle such situation? Shouldn't we add a disclaimer to keep in mind that this is only an example and not always a go-to solution?